libraries/AP_Notify/AP_Notify.cpp

```diff
@@ -78,93 +85,145 @@ AP_Notify::AP_Notify()
 struct AP_Notify::notify_events_type AP_Notify::events;
 
-#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
-#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_PX4_V4
-    PixRacerLED boardled;
-#else
-    AP_BoardLED boardled;
-#endif
-    ToshibaLED_I2C toshibaled;
-    Display display;
-
-#if AP_NOTIFY_SOLO_TONES == 1
-    ToneAlarm_PX4_Solo tonealarm;
-#else
-    ToneAlarm_PX4 tonealarm;
-#endif
+NotifyDevice *AP_Notify::_devices[] = {nullptr, nullptr, nullptr, nullptr, nullptr};
 
-#if AP_NOTIFY_OREOLED == 1
-    OreoLED_PX4 oreoled;
-    NotifyDevice *AP_Notify::_devices[] = {&boardled, &toshibaled, &tonealarm, &oreoled, &display};
-#else
-    NotifyDevice *AP_Notify::_devices[] = {&boardled, &toshibaled, &tonealarm, &display};
-#endif
+// initialisation
+void AP_Notify::init(bool enable_external_leds)
+{
+    
+// Notify devices for PX4 boards
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
+    #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_PX4_V3 // Has enough memory for Oreo LEDs
+        _devices[0] = new AP_BoardLED();
+        _devices[1] = new ToshibaLED_I2C();
+        _devices[2] = new ToneAlarm_PX4();
+        _devices[4] = new Display();
+        
+        // Oreo LED enable/disable by NTF_OLED_THEME parameter
+        if (_oled_theme) {
+            _devices[3] = new OreoLED_PX4(_oled_theme);
+        } else {
+            _devices[3] = nullptr;
+        }
+        
+    #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_PX4_V4 // Has its own LED board
+        _devices[0] = new PixRacerLED();
+        _devices[1] = new ToshibaLED_I2C();
+        _devices[2] = new ToneAlarm_PX4();
+        _devices[3] = nullptr;
+        _devices[4] = new Display();
+        
+    #else   // All other px4 boards use standard devices.
+        _devices[0] = new AP_BoardLED();
+        _devices[1] = new ToshibaLED_I2C();
+        _devices[2] = new ToneAlarm_PX4();
+        _devices[3] = nullptr;
+        _devices[4] = new Display();
+    #endif
 
-#elif CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN
-    ToneAlarm_PX4 tonealarm;
-#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_VRBRAIN_V45
-    AP_BoardLED boardled;
-#else
-    VRBoard_LED boardled;
-#endif
-    ToshibaLED_I2C toshibaled;
-    ExternalLED externalled;
-    NotifyDevice *AP_Notify::_devices[] = {&boardled, &toshibaled, &externalled, &tonealarm};
+    
+// Notify devices for VRBRAIN boards
+#elif CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN  
+    #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_VRBRAIN_V45 // Uses px4 LED board
+        _devices[0] = new AP_BoardLED();
+        _devices[1] = new ToshibaLED_I2C();
+        _devices[2] = new ToneAlarm_PX4();
+        _devices[3] = new ExternalLED();
+        _devices[4] = nullptr;
+        
+    #else
+        _devices[0] = new VRBoard_LED();
+        _devices[1] = new ToshibaLED_I2C();
+        _devices[2] = new ToneAlarm_PX4();
+        _devices[3] = new ExternalLED();
+        _devices[4] = new Display();
```

Comment:
It may be fine but I guess adding the "Display" is a small functional change for the VRBRAIN boards?